From 2afaf39abc461c2729364a6548d37b4227ae5f43 Mon Sep 17 00:00:00 2001
From: Heinrich Toews <heinrich.toews@wago.com>
Date: Mon, 6 Jan 2014 13:43:00 +0000
Subject: [PATCH] omap: 32k-timer: adjust freq value for sysclock to reduce
 drift

omap 32k timer: fix comment

omap: 32k-timer: change log message

Conflicts:
	arch/arm/plat-omap/counter_32k.c
---
 arch/arm/plat-omap/counter_32k.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/plat-omap/counter_32k.c b/arch/arm/plat-omap/counter_32k.c
index e5a6be0..d392dae 100644
--- a/arch/arm/plat-omap/counter_32k.c
+++ b/arch/arm/plat-omap/counter_32k.c
@@ -71,7 +71,19 @@ static void omap_read_persistent_clock64(struct timespec64 *ts)
 	*ts = persistent_ts;
 }
 
+#ifdef CONFIG_MACH_PAC200
+/*
+ * We use internal 32k generation which is based on sys_xtal 26Mhz external
+ * clock which is divided by 800 to provide a 32k like clock.
+ * The theoretical value is 26000000/800 = 32500.
+ * Empirical value is 32498.50669 ~ rounded integer would be 32499.
+ * We use that one. This should give us a drifting of appr. 14 ppm ~ 7.5 min/year.
+ *
+ */
+#define OMAP2_32KSYNCNT_FREQ       32499
+#else
 #define OMAP2_32KSYNCNT_FREQ       32768
+#endif
 
 /**
  * omap_init_clocksource_32k - setup and register counter 32k as a
-- 
2.7.4

